sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractGlobalProtobufFileProcessorTest.java

```diff
@@ -66,125 +71,183 @@ public void prepare() {
+    assertThat(underTest.isGenerated(mockInputFile("generated3"))).isFalse();
+    Map.Entry<String, Charset> expected = entry(toUriString("generated2"), null);
     assertThat(underTest.getRoslynEncodingPerUri()).contains(expected);
   }
 
   @Test
   public void process_generated_new_prop_escaped_csv() throws IOException {
     project1.setProperty("sonar.foo.analyzer.projectOutPaths",
-      "\"" + mockMetadataProtoReport("generated11").toString() + "\",\"" + mockMetadataProtoReport("generated12").toString() + "\"");
-    project2.setProperty("sonar.foo.analyzer.projectOutPaths", mockMetadataProtoReport("generated2").toString());
+      "\"" + mockGenerated("generated11") + "\",\"" + mockGenerated("generated12") + "\"");
+    project2.setProperty("sonar.foo.analyzer.projectOutPaths", mockGenerated("generated2"));
 
     underTest.build(context);
-    assertThat(underTest.getGeneratedFileUris()).containsExactlyInAnyOrder(toUri("generated11"), toUri("generated12"), toUri("generated2"));
-    Map.Entry<URI, Charset> expected = new HashMap.SimpleEntry<>(toUri("generated2"), null);
+    assertThat(underTest.isGenerated(mockInputFile("generated11"))).isTrue();
+    assertThat(underTest.isGenerated(mockInputFile("generated12"))).isTrue();
+    assertThat(underTest.isGenerated(mockInputFile("generated2"))).isTrue();
+    assertThat(underTest.isGenerated(mockInputFile("generated3"))).isFalse();
+    Map.Entry<String, Charset> expected = entry(toUriString("generated2"), null);
     assertThat(underTest.getRoslynEncodingPerUri()).contains(expected);
   }
 
+  @Test
+  public void process_generated_is_not_case_sensitive() throws IOException {
+    project1.setProperty("sonar.foo.analyzer.projectOutPaths", mockGenerated("generated"));
+    project2.setProperty("sonar.foo.analyzer.projectOutPaths", mockGenerated("GENERATED"));
+
+    underTest.build(context);
+    assertThat(underTest.getRoslynEncodingPerUri()).hasSize(1);
+    assertThat(underTest.isGenerated(mockInputFile("GeNeRaTeD"))).isTrue();
+  }
+
   @Test
   public void process_encoding_new_prop() throws IOException {
-    project2.setProperty("sonar.foo.analyzer.projectOutPaths", mockEncodingProtoReport("UTF-8", "encodingutf8").toString());
+    project2.setProperty("sonar.foo.analyzer.projectOutPaths", mockEncoding("UTF-8", "encodingutf8"));
 
     underTest.build(context);
-    assertThat(underTest.getRoslynEncodingPerUri()).containsOnly(entry(toUri("encodingutf8"), StandardCharsets.UTF_8));
-    assertThat(underTest.getGeneratedFileUris()).isEmpty();
+    assertThat(underTest.getRoslynEncodingPerUri()).containsOnly(entry(toUriString("encodingutf8"), StandardCharsets.UTF_8));
+    assertThat(underTest.isGenerated(mockInputFile("encodingutf8"))).isFalse();
+  }
+
+  @Test
+  public void is_not_case_sensitive() throws IOException {
+    project2.setProperty("sonar.foo.analyzer.projectOutPaths", mockGenerated("upper/lower/mixed"));
```

Comment:
I find it confusing that `upper/lower/mixed` are all lower case... shouldn't they be `UPPER/lower/MiXeD` to be consistent with the names?